test/functionalTest/cases/3272_custom_notification_sending_0-length_request/custom_notification_sending_0-length_request.test

```diff
@@ -0,0 +1,314 @@
+# Copyright 2020 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Custom notification sending 0-length request
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create E1/T1
+# 02. Create custom subscription with empty PAYLOAD and URL
+# 03. Create custom subscription with empty PAYLOAD, URL and VERB
+# 04. Create custom subscription with non-empty PAYLOAD, URL, VERB
+# 05. Create custom subscription with empty PAYLOAD, URL and VERB
+# 06. Dump accumulator to see the notification
```

Comment:
None of this steps is actually testing the functionality in this PR, i.e. the `includePayload` field. Thus, I'd suggest the following script of steps:

* 01. Create E1/T1
* 02. Create custom subscription with payload="", includePayload=false
* 03. Create custom subscription with payload="", includePayload=true
* 04. Create custom subscription without payload, includePayload=false
* 05. Create custom subscription without paylaod, includePayload=true
* 06. Create custom subscription without payload="foo", includePayload=false
* 07. Create custom subscription without paylaod="foo", includePayload=true
* 08. Dump accumulator to see the notification
```

Note that VERB is no related with this functionality so it is not needed to change it for the tests. URL is needed always (so there is no need of mentioning it in the step names)

Reply:
Hi @fgalan  sir,

Thank you for your guidance. I have one query redarding test case:
> `# 06. Create custom subscription without payload="foo", includePayload=false`
> `# 07. Create custom subscription without paylaod="foo", includePayload=true`

I think this should be:
> `# 06. Create custom subscription with payload="foo", includePayload=false`
> `# 07. Create custom subscription with paylaod="foo", includePayload=true`

if i have created custom subscription with payload="foo" I am getting the response:

POST http://localhost:REGEX(\d+)/notify
Fiware-Servicepath: /
Content-Length: 3
User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
Ngsiv2-Attrsformat: custom
Host: localhost:REGEX(\d+)
Accept: application/json
Content-Type: text/plain; charset=utf-8
Fiware-Correlator: REGEX([0-9a-f\-]{36})

No JSON object could be decoded=======================================

Reply:
You are right in both cases :)

First one is just a typo.

Second one is due to the accumulator-server.py used by the test tries to decode the payload as a JSON. I don't remember right now if it has some option to process plain text (maybe it has...). However, to keep it simple, maybe we can do:

```
    # 06. Create custom subscription with payload="{ %22x%22: 1}", includePayload=false
    # 07. Create custom subscription with paylaod="{ %22x%22: 1}", includePayload=true
```

%22 is to scape quotes (see https://fiware-orion.readthedocs.io/en/master/user/forbidden_characters/index.html#custom-payload-special-treatment)

Reply:
Hi @fgalan sir,
Thank you for your suggestion. I have updated test cases as per your comment.